src/testing/flow_solver_cases/flow_solver_case_base.cpp

```diff
@@ -26,6 +26,14 @@ void FlowSolverCaseBase<dim,nstate>::display_flow_solver_setup(std::shared_ptr<I
     pcout << "- PDE Type: " << pde_string << std::endl;
     pcout << "- Polynomial degree: " << this->all_param.grid_refinement_study_param.poly_degree << std::endl;
     pcout << "- Final time: " << this->all_param.flow_solver_param.final_time << std::endl;
+
+    this->display_additional_flow_case_specific_parameters(initial_condition);
+}
+
+template <int dim, int nstate>
+void FlowSolverCaseBase<dim,nstate>::display_additional_flow_case_specific_parameters(std::shared_ptr<InitialConditionFunction<dim,nstate,double>> /*initial_condition*/) const
```

Comment:
Is this function really necessary? Why not just override display_flow_solver_setup and just display whatever you want for each case?

Reply:
to avoid repeating code for things that should be displayed by default

Reply:
In that case, display_additional_flow_case_specific_parameters should probably be pure virtual and implemented in every class derived from flow_solver_case_base. `display_flow_solver_setup` should probably not be virtual if it is not meant to be overridden. Also, if you could add each pde_type to `display_flow_solver_setup`, that would probably be good since that will likely make a small bug when someone tries to add a new flow solver case with a new pde type.

Reply:
Addressed in most recent commit